src/alchemlyb/workflows/abfe.py

```diff
@@ -0,0 +1,665 @@
+        elif software.lower() == 'amber':
+            self._extract_u_nk = amber.extract_u_nk
+            self._extract_dHdl = amber.extract_dHdl
+        else: # pragma: no cover
+            raise NameError(f'{software} parser not found.')
+
+    def read(self):
+        '''Read the u_nk and dHdL data from the
+        :attr:`~alchemlyb.workflows.ABFE.file_list`
+
+        Attributes
+        ----------
+        u_nk_list : list
+            A list of :class:`pandas.DataFrame` of u_nk.
+        dHdl_list : list
+            A list of :class:`pandas.DataFrame` of dHdl.
+        '''
+        u_nk_list = []
+        dHdl_list = []
+        for xvg in self.file_list:
+            try:
+                u_nk = self._extract_u_nk(xvg, T=self.T)
+                self.logger.info(
+                    f'Reading {len(u_nk)} lines of u_nk from {xvg}')
+                u_nk_list.append(u_nk)
+            except: # pragma: no cover
+                self.logger.warning(
+                    f'Error reading read u_nk from {xvg}.')
+
+            try:
+                dhdl = self._extract_dHdl(xvg, T=self.T)
+                self.logger.info(
+                    f'Reading {len(dhdl)} lines of dhdl from {xvg}')
+                dHdl_list.append(dhdl)
+            except: # pragma: no cover
+                self.logger.warning(
+                    f'Error reading read dhdl from {xvg}.')
+
+        # Sort the files according to the state
+        if len(u_nk_list) > 0:
+            self.logger.info('Sort files according to the u_nk.')
+            column_names = u_nk_list[0].columns.values.tolist()
+            index_list = sorted(range(len(self.file_list)),
+                key=lambda x:column_names.index(
+                    u_nk_list[x].reset_index('time').index.values[0]))
+        else:
+            self.logger.info('Sort files according to the dHdl.')
+            column_names = sorted([dHdl.reset_index('time').index.values[0]
+                                   for dHdl in dHdl_list])
+            index_list = sorted(range(len(self.file_list)),
+                key=lambda x:column_names.index(
+                    dHdl_list[x].reset_index('time').index.values[0]))
+
+        self.file_list = [self.file_list[i] for i in index_list]
+        self.logger.info("Sorted file list: \n %s", '\n'.join(self.file_list))
+        self.u_nk_list = [u_nk_list[i] for i in index_list]
+        self.dHdl_list = [dHdl_list[i] for i in index_list]
+        self.u_nk_sample_list = None
+        self.dHdl_sample_list = None
+
+    def run(self, skiptime=0, uncorr='dhdl', threshold=50,
+            methods=('mbar', 'bar', 'ti'), overlap='O_MBAR.pdf',
+            breakdown=True, forwrev=10, *args, **kwargs):
+        ''' The method for running the automatic analysis.
+
+        Parameters
+        ----------
+        skiptime : float
+            Discard data prior to this specified time as 'equilibration' data. Units
+            picoseconds. Default: 0.
```

Comment:
How do you know that the input data are in ps? I think the only thing you can say here is that it's the same unit that was stored by the code that wrote it. alchemlyb does not really know about time.

Reply:
@orbeckst I think The u_nk pandas data frame has one column which is time?

Reply:
Yes, but we have no idea what the unit of this time column is. You don't need code changes, but remove "Units picoseconds" and replaces with "The unit of time is the same as in the output from the package that generated the time series of dH/dl and u_nk values."